### ISSUE TITLE
Feature Request: Ability to Archive Chats

### ISSUE THREAD

**Issue by mxmerz** [peter-iakovlev/Telegram#13]
Hi, I would very much appreciate the ability to archive certain chats:
**Problem:** I have a long chat list in which a lot of chats are not active anymore – but I don’t want to delete the chat history.
**Proposed solution:** Ability to archive chats. For examples on how this feature could be implemented, see Signal on iOS or WhatsApp.

**Comment by Kutsan** [comment]
+1

**Comment by naibaf0** [comment]
+1

**Comment by nicholaschum** [comment]
+1

**Comment by adrianbona** [comment]
+1

**Comment by stelladauer** [comment]
+1

**Comment by danieltvela** [comment]
+1

**Comment by pythoninthegrass** [comment]
+1000. How is it that [Whatsapp](https://www.whatsapp.com/faq/en/bb/20888029), [Hangouts](https://support.google.com/hangouts/answer/3112142?co=GENIE.Platform%3DDesktop&hl=en), [Messenger](https://www.facebook.com/help/messenger-app/1508565246032840), and a litany of other proprietary closed-source programs have this basic feature and Telegram doesn't??

**Comment by baglio** [comment]
+1

**Comment by Brackmann** [comment]
+1

**Comment by poulenque** [comment]
+1

**Comment by phisco** [comment]
+1

**Comment by hawk93** [comment]
+1

**Comment by MrAnas** [comment]
+1

**Comment by joezeng** [comment]
+1

**Comment by mjapple** [comment]
+1

**Comment by abpasti** [comment]
+1

**Comment by erny** [comment]
+1

**Comment by abpasti** [comment]
+1

**Comment by AdrienCardoso** [comment]
+1 Please do this.  I am an early adopter and has a really long chat list.

**Comment by mxmerz** [comment]
Version 3.11 introduces Archived Stickers – are archived chats planned in the future? Is anybody working on Telegram even looking at the issues here on GitHub?

**Comment by luansmm** [comment]
+1

**Comment by joezeng** [comment]
+1

**Comment by Brackmann** [comment]
PLEASE GIVE US A FEEDBACK

**Comment by gkope** [comment]
+1, archive please 😄

**Comment by shailensobhee** [comment]
+2,000,000 We want this feature!

**Comment by Doelli** [comment]
+1

**Comment by dbuenor** [comment]
+1

**Comment by jamespadfoot** [comment]
+1

**Comment by craaaa** [comment]
+1

**Comment by bgta** [comment]
+1

**Comment by ryukinix** [comment]
oooooooohhhh man, give to us some feedback!!!

**Comment by samttau** [comment]
+1 Surely this is more important than masks?

**Comment by yoavprat** [comment]
+1

**Comment by jonaselan** [comment]
+1

**Comment by mlwane** [comment]
+1

**Comment by danielfireman** [comment]
+1

**Comment by Brackmann** [comment]
Well... you can pin conversations now, but still it is not what we are asking for :(

**Comment by gimmy** [comment]
+π

**Comment by gaurges** [comment]
+100500

**Comment by meslahik** [comment]
+1
Why not considering this handy and important feature?!

**Comment by gkope** [comment]
Is this repository really used? no replys...

**Comment by ChristoferKarlsson** [comment]
+1

**Comment by hiccups1980** [comment]
+1

**Comment by phplego** [comment]
+1

**Comment by tjhorner** [comment]
The official channel for feature requests is through Settings, under "Ask a question". I don't think that creating an issue here will get it to the dev.

**Comment by SannePetersNL** [comment]
+1

**Comment by fkosec** [comment]
+1

**Comment by ericswpark** [comment]
+1

**Comment by kossmos** [comment]
+1

**Comment by afdallah** [comment]
+1

**Comment by alcatian** [comment]
+1000

**Comment by afdallah** [comment]
+1

**Comment by rmsa5** [comment]
+1 . Still "no one assigned" on such a basic must-have feature ?

**Comment by awesome-Leni** [comment]
+1

*also it could be interesting to hear what's the motivation behind the idea of not developing this feature

**Comment by agtokty** [comment]
+1

**Comment by rmsa5** [comment]
+1 classic case of zerofucksgiven it would seem

**Comment by maltalef** [comment]
+1

**Comment by abzubarev** [comment]
+1

**Comment by SannePetersNL** [comment]
+1

**Comment by masc3d** [comment]
as `Clear History` doesn't seem to offer much over `Delete` except keeping the conversation window open it could be possibly replaced with `Archive` which would instead remove the conversation from the sidebar while  keeping the history in case the conversation is re-opened via contacts/archive.

**Comment by esi-bln** [comment]
+1
one of the reason I still haven't migrated all my conversations from whatsapp. I'm really wondering why this is still not being addressed

**Comment by olea** [comment]
+1

**Comment by plakhin** [comment]
+1
please add this feature

**Comment by carlosmjr** [comment]
+1

**Comment by shailensobhee** [comment]
+1
This should be the hottest next feature to be implemented!!

**Comment by gmeszaros** [comment]
+1

**Comment by RaydenX93** [comment]
+1

**Comment by brunohq** [comment]
+1

**Comment by JohanTG** [comment]
+1

**Comment by tiagobabo** [comment]
+1

**Comment by AdonisZA** [comment]
+1 
Yes please - this is such an important feature. Please integrate asap if possible.

**Comment by markalexandercastillo** [comment]
+1

**Comment by jesmg** [comment]
+1

**Comment by dilkROM** [comment]
+1

**Comment by torresweb** [comment]
+1

**Comment by zenhaeus** [comment]
+1

**Comment by dublado** [comment]
+1

**Comment by rinukkusu** [comment]
+1

**Comment by div4d** [comment]
+1

**Comment by shailensobhee** [comment]
Can we please make this the next hottest feature to implement?

**Comment by olea** [comment]
> Can we please make this the next hottest feature to implement?

Hopefuly would be easier to implement than anything made with a blockchain ;-D

**Comment by rsalinas** [comment]
I really need this.  My chat list is so cluttered without this! Don't let WhatsApp overtake you ;) 

It should be very easy to implement: 

- for every chat there is a flag "Archived", defaulting to false.  
- A new button will set the flag. 
- The standard view filters out those chats. 
- Any addition will change this flag.  
- You simply need an extra view "Archived chats".  

It should be fairly easy for such good developers :)) 

Thanks!

**Comment by div4d** [comment]
I think that Telgram will be better if there will be a primary and a secondary chat list. Users can move to the secondary swiping left over the primary one (the opposite swipe for the "setting panel".
In this secondary chat list users can archive some old chat or some useful but annoying channel, or whatever.
The main scope is to keep the primary chat list as clean as possible

**Comment by Momentanius** [comment]
+1 ! I want this. And if possible, make archived chats/groups that are silenced stay archived!!

**Comment by Zwimber** [comment]
+1

**Comment by osfast** [comment]
+1

**Comment by Hugopiovesan** [comment]
+1
Come on Telegram, its been years now. Hear your community!

**Comment by ghost** [comment]
I'm SAD to know this has been requested for 3 years now and no implementation. Look at the competition, they have it! I don't want to delete a chat history, but I have a friend who just died and I either delete everything and erase my memories, or keep staring at her photo. True story.

**Comment by Younes-L** [comment]
Why don't y'all use tabs

**Comment by ghost** [comment]
> Why don't y'all use tabs

Can you elaborate?

**Comment by Brackmann** [comment]
> Why don't y'all use tabs

This is not what we're looking for.
It lacks features and usability.

**Comment by Younes-L** [comment]
@Brackmann @Rosseiro  you can set a tab for favourites == unarchived chats, and hide the everything tab (archived + unarchived)

It's almost the same, you won't be spending much time looking at the archived chats anyway.

You can even have a tab for unread chats where only those with unread badge will show up !

**Comment by Brackmann** [comment]
There is no such option in Telegram. Could you please give us some help?
Thanks in advance.

**Comment by iamGavinJ** [comment]
+1

**Comment by zahedim** [comment]
+1

**Comment by ghost** [comment]
I'm checking back here just to tell you this is actually an urgent request. Telegram is currently my only means of communicating with the world. And I have to stare at my dead friend's chat and photo **everytime** I launch the app. As I said before, deleting the conversation is not an option, because I want to remember her. Is it really that difficult?

By the way, I'm sorry people are too stupid to prefer WhatsApp over Telegram. They call me "the willingfully different one" for preferring Telegram.

**Comment by rmsa5** [comment]
@Rosseiro seriously this is no place to unload such private matter. Just install one of telegram's desktop client or even access the web client, export / copy-paste that whole damn conversation to a offline word document or google doc, print it and/or archive it in a safe place, grieve as long as you need to, and remove the telegram conversation (it's not like you're going to pick up the conversation again any time soon anyway...)

**Comment by olea** [comment]
Anyhow, this is a demanded feature, probably easy to implement for developer team give no attention.

I wonder if any team member reads this issue indeed :-m

**Comment by shailensobhee** [comment]
@peter-iakovlev Can you please kindly shortly explain to all of us how new stickers have higher priority than this super mega hot (basic) feature request that's been open for over 2 years?

**Comment by jacksonalexandre** [comment]
+1

**Comment by oscon-it** [comment]
+1 three years ago I needed this feature, I still need it today. Could someone please suggest something?

**Comment by hgouveia** [comment]
+1

**Comment by rmsa5** [comment]
@oscon-it sadly, I ended up giving up, and going back to whatsapp, which is a pain in the ass : no real multi-platform support (acting as a remote to phone app doesn't count), but at least you can archive. :(

**Comment by Brackmann** [comment]
Is there another place to post this issue? I don't see any contributions, commits and replies from Peter Iakovlev in his github account. Any suggestions?

**Comment by ghost** [comment]
@Brackmann true, I believe there is no way this is the only medium to give feedback for an application used by hundreds of million people, and a lot superior to WhatsApp. But in this particular issue, WhatsApp is on the lead. We NEED the ability to archive chats!

**Comment by zydone** [comment]
I can give up the archive feature in telegram , than lose all my chats and history when whatsapp decides to ban without a reason a number , I own a company and cause of the simple fact that I do business with my whatsapp number I am consider to be a spam , I deal with 100 to 150 customers daily and they do not allow business transactions on whatsapp be carefull with whatsapp

**Comment by kuzyakiev** [comment]
+1,
really need this feature. It's cool realized in facebook messenger.
Why I need it?
To keep archived conversations that i read - and it no need my attention or actions from me. And leave active conversation where I need back with reply or other kind of attention. 

What I can do to make this feature happens?

**Comment by nunesbeto** [comment]
+1

**Comment by leogouveia** [comment]
+1

**Comment by mansouryaacoubi** [comment]
How many years should people continue asking for a simple feature, without even getting an answer or reaction? In coming november it'll be 3 years and yet no answer.
@TelegramOrg 
Would be nice just to know whether this will be planned or declined.

**Comment by ghost** [comment]
I was about to say the same thing. Please, at least end our anxiety. I didn't lie when I said I have a deceased friend show up all the time. It is painful.

**Comment by oscon-it** [comment]
Well... is this the right place to ask for this *necessary* feature? I am starting to doubt....

**Comment by paul-grassl** [comment]
+1
Is there any other way then this or asking a question to the volunteer support to draw attention to this matter?

**Comment by oscon-it** [comment]
Hey guys: *THEY ADDED* the feature with the last update!

https://telegram.org/blog/export-and-more

It is called “Chat export tool”, I haven’t tried it yet, but it seems good, except for the fact that there is a size limit.

Do this feature provide what was asked here?

**Comment by paul-grassl** [comment]
It seems to me that this is more of a feature to save your chats on your computer (maybe to comply with GDPR or so). What would be great is a simple Archive folder at the bottom of your chat list where you can easily put all chats you don't need atm. Just like WhatsApp has it.

**Comment by ChristoferKarlsson** [comment]
Export is a really nice feature! Hoping this will be more of a standard feature among chat clients in the future. However, at least in my case, I don't want to delete the chats from Telegram. Just hide them from the chat list. I still want to be able to read and search old archived chats. When an archived chat gets active again, I really would like to see the past conversation in Telegram. If you use export and then clear the chat, you would have to look up one or multiple exported chat files to find the past conversation.

**Comment by ghost** [comment]
I'm using the Windows version and I see nothing that resembles the so called archive feature. Like Christofer said, we don't want to remover the conversations, only hide them.

**Comment by olea** [comment]
> Do this feature provide what was asked here?

No. It can't archive conversations.

**Comment by kriskhaira** [comment]
If it helps the maintainers, this is a good case for having this feature. I actually practice InboxZero with not just my email but IM too, which means I try to keep my WhatsApp inbox empty everyday.
https://www.lifehack.org/articles/lifehack/ultimate-way-inbox-zero.html
https://www.entrepreneur.com/article/290175

I'm guessing a lot of people here may be doing this, without knowing there's a name for this.

Obviously without this feature, my inbox is very messy in Telegram, which means I avoid using it as much as possible and only have with 1–2 threads there which are usually groups with people who insist on using Telegram. I'd gladly move all my conversations with people from WhatsApp to Telegram if it had an Archive feature.

**Comment by periflo** [comment]
Please add this! I think it's a really useful feature which helps reducing clutter.

**Comment by jelur** [comment]
This will decultter my Telegram a lot... please!

**Comment by rmsa5** [comment]
I don't know about you guys, but I gave up on telegram. A lot of potential, but too much of a mess in the long run.

**Comment by zydone** [comment]
Hi there I gave up on Rambox , I got my whatsapp number cancel , my facebook block , and everything I have on rambox the social networks do not like the service , I had reported this before but I guess nothing was done , I have to work with all the social networks and they advise me to remove my account from the rambox system that they do not supported

**Comment by waaronking** [comment]
It's 2019 and I just wanted to add a +1 to this

**Comment by christianalai** [comment]
It's been 4 years since this issue was created. Are there any plans to develop the archive feature?

**Comment by shailensobhee** [comment]
Sorry guys. Stickers are more important than the hottest feature you all
need. - Devs

**Comment by martijnw** [comment]
Driving me nuts not having a simple ' archive chat'  feature. So many messages that I have started to use whatsapp again. There is no way to organize and archive chats without directly deleting them on Telegram. This feature should have been there at least 4 years ago! please fix, this is just annoying.

**Comment by martijnw** [comment]
> Do this feature provide what was asked here?

No. It does not... Export chat, not the same as archive chat. Can't beleive it is so hard to do. Giving up on Telegram. After 4 years of waiting for a basic feature, I waited enough. The app is nice, bots are nice, groups are great and whoop whoop, it has got stickers but a simple archive chat feature is missing, causes many frustrations with my sometimes 1000s of open messages that I can no longer stand it and want my sanity back. UNINSTALL

**Comment by momper14** [comment]
+1 for this feature. Don’t know why it is so hard to implement.

**Comment by albertnev** [comment]
I've got my chat history full of chats I no longer want to have in there, but don't want to lose the messages and files. Being able to archive them would be lovely, as they would be "collapsed" without losing anything and having much more visual space. I don't know why this hasn't been implemented yet.

**Comment by mansouryaacoubi** [comment]
Well I guess this feature will never come to telegram. It's now been nearly 4 years with no reaction to this desirable enhancement. Therefore I think that this will be only a WhatsApp feature.

**Comment by olea** [comment]
> Therefore I think that this will be only a WhatsApp feature.

Correction: Signal has it too :-)

**Comment by mansouryaacoubi** [comment]
Oh yeah sorry... I mean.. nearly every messenger has this ability except for telegram :D

**Comment by ghost** [comment]
So, I believe it is now safe to assume this will never be implemented? I don't know how this is possible because the app is in active development! Is it the real truth that the devs just think this feature is cosmetic?

**Comment by jelur** [comment]
> So, I believe it is now safe to assume this will never be implemented? I don't know how this is possible because the app is in active development! Is it the real truth that the devs just think this feature is cosmetic?

Themes is a cosmetic feature, and is fully developed

**Comment by ShadowJonathan** [comment]
Telegram made a mistake by leaving their mobile app development into the hands of private people, who dont even listen to the community.

**Comment by oscon-it** [comment]
We've been waiting for 4 years. This is a basic feature. Dozen of players have described why it is necessary. Is this enough?

**Comment by oscon-it** [comment]
We've been waiting for 5 years. This is a basic feature. Dozen of user have described why it is necessary. Is this enough?

**Comment by oscon-it** [comment]
We've been waiting for 5 years. This is a basic feature. Dozen of users have described why it is necessary. Is this enough?

**Comment by Younes-L** [comment]
@rosseiro Durov just tweeted about it, said it's coming. https://mobile.twitter.com/durov/status/1122509203996909571
I missed the previous tweet https://mobile.twitter.com/durov/status/1101067184988667905 where he was asking about the utility of archive 😂
"Why do you people want this feature ? What do you want to do with it ?"
Lmao smbh

**Comment by ghost** [comment]
Ahaha, the utility? Anyways, it's good to see it happening. I don't know if he's reading here, but THANK YOU Durov!

**Comment by rmsa5** [comment]
He's got to be trolling 😂
This guy is genius for having built telegram, but he sounds like a caveman asking about archiving usecases... Smell of irony.

**Comment by christianalai** [comment]
It's [here](https://telegram.org/blog/archive-and-new-design) finally!  🎉

**Comment by Hugopiovesan** [comment]
> It's [here](https://telegram.org/blog/archive-and-new-design) finally! 🎉

Oh my god. I cant believe its actually done! After 4 years they finally did it!

**Comment by periflo** [comment]
Everybody rejoice!
